src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiAdapterFactoryProcessor.java

```diff
@@ -131,6 +163,47 @@ private ExecutableElement getJsonAdapterMethod(Element element) {
     return null;
   }
 
+  private void error(Element element, String message, Object... args) {
+    if (args.length > 0) {
+      message = String.format(message, args);
+    }
+    processingEnv.getMessager().printMessage(ERROR, message, element);
+  }
+
+  private boolean containsJsonAdapterFactoryMirror(List<? extends TypeMirror> interfaces) {
```

Comment:
In other places we check for implementation of an interface (Parcel extension, in particular), we have to do this recursively for parent types of each implemented interface/extended class, in case the user has something like a `BaseJsonAdapterFactory implements JsonAdapter.Factory`.  I wonder if that's even a reasonable use case here?